### DISCUSSION TITLE
A question about `basePath`

### DISCUSSION THREAD

**Discussion by SiarheiBobryk** [gregrickaby/nextjs-github-pages#61] [General]
Hey, @gregrickaby!

First of all, BIG thanks for your detailed instructions and the yml-file. I've been looking for a good starting boilerplate for deploying a Nextjs App via GitHub Pages and finally found it. You've helped me a lot.

Allow me to ask you a question.

I followed your manual in README.md but did not manage to make it. There was something wrong the path to images. Then I started experimenting. Eventually, the following changes worked for me:

1. `next.config.mjs`:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  //...
  basePath: "", // just empty string, but not the repo name
  //...
};

export default nextConfig;
```

2. Not applying any changes to the `src` prop of `Image`.

See my working [commit](https://github.com/SiarheiBobryk/SiarheiBobryk.github.io/commit/f23cb444284c3c03623d5d4eda0b1e69d3701208#diff-18c049b08c4a0f5ab451c598aeb2c4848bb9d7877b51ca3e5effb94a225814d2R15)

I'm trying to understand how it works, but I cannot get why the method above works for my repo? Maybe they've changed something in the API. Please explain.

Thank you!

**Comment by ChuckmonsterXlx** [comment]
Surely because you are using the unique repository provided by GitHub 'name.github.io,' that is the reason why it is not necessary to add the 'basepath.'

In case you have another different repository, you would need to add to the basepath the name of your repository as @gregrickaby explains.

Greetings!

**Comment by SiarheiBobryk** [comment]
Thanks for answering, @ChuckmonsterXlx!

One small clarification.

For `https://github.com/<MyGitHubID>/<MyGitHubID>.github.io` repos (we can create only one instance of it per GitHub account), a base path is basically `/` (root). We don't need to specify it.

For the rest of repos (for example, `https://github.com/<MyGitHubID>/my-new-project`) I'd have to specify the base path. In this case, it would be `/my-new-project`.

Correct?

**Comment by ChuckmonsterXlx** [comment]
It's right!

**Comment by SiarheiBobryk** [comment]
Thanks again, @ChuckmonsterXlx!
